src/hotspot/share/gc/shenandoah/shenandoahHeapRegion.hpp

```diff
@@ -370,19 +370,28 @@ class ShenandoahHeapRegion {
   inline HeapWord* allocate_fill(size_t word_size);
 
   inline void clear_live_data();
-  void set_live_data(size_t s);
-
-  // Increase live data for newly allocated region
-  inline void increase_live_data_alloc_words(size_t s);
+  void set_live_data_after_fullgc(size_t s, ShenandoahMarkingContext* context, size_t index);
 
   // Increase live data for region scanned with GC
   inline void increase_live_data_gc_words(size_t s);
+  inline bool has_marked() const;
+
+  inline bool has_live(ShenandoahMarkingContext* context, size_t index) const;
+
+  // Returns bytes identified as live by most recently completed marking effort.  Can only be called during safepoints.
```

Comment:
```most recently completed marking effort``` : 

Three questions:
1. I assume we here mean an old marking cycle for an old region, and a young marking cycle for a young region? If so, let's explicitly state that.
2. what does this return if called before a completed marking. 
3. what does this return when a marking is in progress? Does it return the value from a previously completed marking effort, or something else?